### PR TITLE
Move time info to file manager and remove unused entries

### PR DESCRIPTION
**Pull request by bartnijssen**
@arbennett - this one needs a somewhat more thorough review. If you have major edits, just make a PR to the aww/time_info branch in github.com/bartnijssen/summa

### CODE REVIEW

File: build/source/driver/summa_init.f90 [hunk 1]
```diff
@@ -65,7 +65,7 @@ subroutine summa_initialize(summa1_struc, err, message)
  USE summa_type, only:summa1_type_dec                        ! master summa data type
  ! subroutines and functions: initial priming
  USE summa_util, only:getCommandArguments                    ! process command line arguments
- USE summaFileManager,only:summa_SetDirsUndPhiles            ! sets directories and filenames
+ USE summaFileManager,only:summa_SetTimesDirsUndPhiles       ! sets directories and filenames
```
Comment (arbennett) [inline build/source/driver/summa_init.f90:68]: Let's rename this now while it's being changed - `summa_SetTimesDirsAndFiles`

File: build/source/driver/summa_init.f90 [hunk 2]
```diff
@@ -183,7 +183,7 @@ subroutine summa_initialize(summa1_struc, err, message)
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
 
  ! set directories and files -- summaFileManager used as command-line argument
- call summa_SetDirsUndPhiles(summaFileManagerFile,err,cmessage)
+ call summa_SetTimesDirsUndPhiles(summaFileManagerFile,err,cmessage)
```
Comment (arbennett) [inline build/source/driver/summa_init.f90:186]: Same rename as above.

File: build/source/hookup/summaFileManager.f90 [hunk 3]
```diff
@@ -19,143 +19,122 @@
...
-MODULE summafilemanager
+MODULE summaFileManager
 use nrtype
 implicit none
 public
 ! summa-wide pathlength
-integer(i4b),parameter::summaPathLen=4096
+integer(i4b),parameter       :: summaPathLen=4096
+! defines the time of the run
+CHARACTER(LEN=summaPathLen)  :: CONTROL_VRS      = 'SUMMA_FILEMANAGER_V2.0'         ! control version
```
Comment (arbennett) [inline build/source/hookup/summaFileManager.f90:32]: Change to `SUMMA_FILEMANAGER_V3.0.0`

File: build/source/hookup/summaFileManager.f90 [hunk 4]
```diff
@@ -19,143 +19,122 @@
...
+CHARACTER(LEN=summaPathLen)  :: MODEL_INITCOND   = 'summa_zInitialCond.txt'         ! model initial conditions
+CHARACTER(LEN=summaPathLen)  :: PARAMETER_TRIAL  = 'summa_zParamTrial.txt'          ! trial values for model parameters
+CHARACTER(LEN=summaPathLen)  :: OUTPUT_PREFIX    = 'summa_output_'                  ! prefix for the output file
+
 contains
 
+ ! **************************************************************************************************
+ ! public subroutine summa_SetTimesDirsUndPhiles: Sets times, directories and filenames for summa run
+ ! **************************************************************************************************
+ subroutine summa_SetTimesDirsUndPhiles(summaFileManagerIn,err,message)
```
Comment (arbennett) [inline build/source/hookup/summaFileManager.f90:56]: Rename to `summa_SetTimesDirsAndFiles`

File: build/source/hookup/summaFileManager.f90 [hunk 5]
```diff
@@ -19,143 +19,122 @@
...
-  err=20; return
-end if
-! read information from file
-ierr=0 ! initialize errors
+ ! input/output vars
+ character(*),intent(in)              :: summaFileManagerIn
+ integer(i4b),intent(out)             :: err
+ character(*),intent(out)             :: message
+ ! local vars
+ character(*),parameter               :: summaFileManagerHeader="SUMMA_FILE_MANAGER_V2.0"
```
Comment (arbennett) [inline build/source/hookup/summaFileManager.f90:70]: Rename to `SUMMA_FILE_MANAGER_V3.0.0`

File: build/source/hookup/summaFileManager.f90 [hunk 6]
```diff
@@ -19,143 +19,122 @@
...
+ integer(i4b)                         :: unt            ! file unit (free unit output from file_open)
+ character(LEN=linewidth),allocatable :: charline(:)    ! vector of character strings
+ integer(i4b)                         :: iControl, nControl       ! number of model info
+ character(len=summaPathLen)          :: varEntry          ! name of model info
+ character(len=32)                    :: option         ! option for model info
 
-call readLine(fileUnit,SETNGS_PATH,     err,message); if(err/=0)return
-call readLine(fileUnit,INPUT_PATH,      err,message); if(err/=0)return
-call readLine(fileUnit,OUTPUT_PATH,     err,message); if(err/=0)return
+ err=0; message="summa_SetTimesDirsUndPhiles/"
```
Comment (arbennett) [inline build/source/hookup/summaFileManager.f90:81]: Rename to `summa_SetTimesDirsAndFiles`

File: build/source/hookup/summaFileManager.f90 [hunk 7]
```diff
@@ -19,143 +19,122 @@
...
+ write(runinfo_fileunit,*) 'Run start time on system:  ccyy='//cdate(1:4)//' - mm='//cdate(5:6)//' - dd='//cdate(7:8), &
+                         ' - hh='//ctime(1:2)//' - mi='//ctime(3:4)//' - ss='//ctime(5:10)
+ close(runinfo_fileunit)
 
-do
- ! read line that is not comment
- read(fileUnit,*) inputString
- if (inputString(1:1) /= '!') exit
-end do
+ end subroutine summa_SetTimesDirsUndPhiles
```
Comment (arbennett) [inline build/source/hookup/summaFileManager.f90:137]: Rename to `summa_SetTimesDirsAndFiles`

File: docs/input_output/SUMMA_input.md [hunk 8]
```diff
@@ -10,62 +10,94 @@ SUMMA input files are either ASCII format or NetCDF. The general characteristics
...
+
+The following items must be provided in the master configuration file. Order is not important, as the entries are each associated with a keyword.  Each keyword and entry pair must be on its own line, but may be followed by a comment (started by the '!' character), and you can add lines of comments between the items. Each entry must be enclosed in single quotes `'entry'`. The associations of the keywords to the actual variable name that is used in the SUMMA source code can be found in `summaFileManager.f90`, along with its default value where appropriate.
+
+`controlVersion`: Version of the file manager that should be used to process the master configuration file. At this time, this string should be equal to `'SUMMA_FILE_MANAGER_V2.0'`.  Note, this version of the code is not backward compatible with versions using `SUMMA_FILE_MANAGER_V1.0`.
```
Comment (arbennett) [inline docs/input_output/SUMMA_input.md:25]: Change `SUMMA_FILE_MANAGER_V2.0` to `SUMMA_FILE_MANAGER_V3.0.0`

File: build/source/driver/summa_init.f90 (hunk 1 above)
Comment (bartnijssen) [reply build/source/driver/summa_init.f90:68]: Done also renamed `Philes` to `Files`

File: build/source/driver/summa_init.f90 (hunk 2 above)
Comment (bartnijssen) [reply build/source/driver/summa_init.f90:186]: Same

File: build/source/hookup/summaFileManager.f90 (hunk 4 above)
Comment (bartnijssen) [reply build/source/hookup/summaFileManager.f90:56]: Done

File: build/source/hookup/summaFileManager.f90 (hunk 3 above)
Comment (bartnijssen) [reply build/source/hookup/summaFileManager.f90:32]: Done

File: build/source/hookup/summaFileManager.f90 (hunk 5 above)
Comment (bartnijssen) [reply build/source/hookup/summaFileManager.f90:70]: Done

File: build/source/hookup/summaFileManager.f90 (hunk 6 above)
Comment (bartnijssen) [reply build/source/hookup/summaFileManager.f90:81]: Done

File: build/source/hookup/summaFileManager.f90 (hunk 7 above)
Comment (bartnijssen) [reply build/source/hookup/summaFileManager.f90:137]: Done

File: docs/input_output/SUMMA_input.md (hunk 8 above)
Comment (bartnijssen) [reply docs/input_output/SUMMA_input.md:25]: Done

**Comment by andywood** [comment]
Nice, thanks Andrew and Bart.

I had left Martyn's 'DirsUndPhiles' as a distinctive Clark/Kavetski-ism
...(I think that chunk of code originally came from some clark/dmitri
work).  Ok by me to normalize it though ...

**Comment by andywood** [comment]
major sub-part of PR #386

**Pull request merged**